src/combinators/mod.rs

```diff
@@ -1020,4 +1035,16 @@ mod test {
         }
         assert_eq!(&buf, &[0, 0, 0, 4, 't' as u8, 'e' as u8, 's' as u8, 't' as u8]);
     }
+
+    #[test]
+    fn test_back_to_the_buffer() {
+        let mut buf = [0; 8];
+        let rest = buf.reserve_write_use(
+            4,
+            move |buf| string("test")(WriteCounter::new(buf)).map(|counter| counter.into_inner()),
+            move |buf, len| be_u32(len as u32)(buf)
+        ).unwrap();
+        assert_eq!(&rest, &[]);
```

Comment:
Might make sense to test in the test that the rest is not just the empty slice, but that it's at the end of the slice above. E.g. make the slice one byte bigger and write a `be_u8` at the end and check if that's also in the final output at the right place (in the very end and not overwriting something else) later.

Reply:
Makes sense, will do